collie/interactions/datasets.py

```diff
@@ -658,6 +673,12 @@ def __init__(self,
                  num_items: int = 'infer',
                  seed: Optional[int] = None,
                  shuffle: bool = False):
+        if negative_sample_type not in ('item', 'user'):
+            raise ValueError(
+                '``negative_sample_type`` must be either ``item`` or ``user``, not '
+                f'{negative_sample_type}!'
+            )
```

Comment:
Since we use this in multiple classes, my vote is to pull it out into a helper function in this file!